tests/Backend/SOX/MergeRequestsTest.php

```diff
@@ -730,6 +731,135 @@ public function testConfigurationUpdatedInBranch(): void
+        $devBranchComponents->updateConfiguration((new Configuration())
+            ->setComponentId($componentId)
+            ->setConfigurationId($configurationId)
+            ->setName('Main updated')
+            ->setDescription('First update description')
+            ->setConfiguration(['main' => 'update'])
+            ->setChangeDescription('Update config')
+            ->setIsDisabled(true));
+
+        $configInDev = $devBranchComponents->getConfiguration($componentId, $configurationId);
+        $this->assertSame('update', $configInDev['configuration']['main']);
+        $this->assertSame(2, $configInDev['version']);
+        $this->assertSame('Update config', $configInDev['changeDescription']);
+
+        $devBranchComponents->updateConfiguration((new Configuration())
+            ->setComponentId($componentId)
+            ->setConfigurationId($configurationId)
+            ->setName('second main updated')
+            ->setDescription('last update desc')
+            ->setChangeDescription('last update')
+            ->setConfiguration(['main' => 'update again']));
+
+        $configState = (new ConfigurationState())
+            ->setComponentId($componentId)
+            ->setConfigurationId($configurationId)
+            ->setState(['dev-branch-state' => 'state'])
+        ;
+
+        $devBranchComponents->updateConfigurationState($configState);
+
+        $configInDev = $devBranchComponents->getConfiguration($componentId, $configurationId);
+        $this->assertSame('update again', $configInDev['configuration']['main']);
+        $this->assertSame(3, $configInDev['version']);
+        $this->assertSame('last update', $configInDev['changeDescription']);
+        $this->assertSame(['dev-branch-state' => 'state'], $configInDev['state']);
+
+        $lastVersionIdentifierInDevBranch = $configInDev['currentVersion']['versionIdentifier'];
+
+        $newConfigId = $this->generateUniqueNameForString('new-config');
+        $devBranchComponents->addConfiguration(
+            (new Configuration())
+                ->setComponentId($componentId)
+                ->setConfigurationId($newConfigId)
+                ->setConfiguration(['main' => 'created in branch'])
+            ->setName($this->generateDescriptionForTestObject()),
+        );
+
+        $this->initEvents($this->getDefaultBranchStorageApiClient());
+
+        $mrId = $this->developerClient->createMergeRequest([
+            'branchFromId' => $newBranch['id'],
+            'branchIntoId' => $defaultBranch['id'],
+            'title' => $this->generateUniqueNameForString('mr-title'),
+            'description' => $this->generateUniqueNameForString('mr-description'),
+        ]);
+
+        $reviewerClient = $this->getReviewerStorageApiClient();
+        $this->developerClient->mergeRequestRequestReview($mrId);
+        $mr = $reviewerClient->getMergeRequest($mrId);
```

Comment:
proc to tady musi getovat ten reviewer? jakoze asi to nicemu nevadi, ale unika mi neco?

Reply:
Jen copy paste odjinud